extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -254,17 +254,29 @@ export const popupMessageListener = (request: Request, sessionStore: Store) => {
     password,
     keyPair,
     imported = false,
+    isSettingHashKey = false,
   }: {
     mnemonicPhrase: string;
     password: string;
     keyPair: KeyPair;
     imported?: boolean;
+    isSettingHashKey?: boolean;
   }) => {
     const { publicKey, privateKey } = keyPair;
 
     const allAccounts = allAccountsSelector(sessionStore.getState());
     const accountName = `Account ${allAccounts.length + 1}`;
 
+    let activeHashKey = await getActiveHashKeyCryptoKey({ sessionStore });
+    if (activeHashKey === null && isSettingHashKey) {
```

Comment:
I'm adding an explicit flag here just to avoid any weird edge cases where a user gets to this block of code and unexpectedly doesn't have their hash key already. Deriving a new hash key would cause a user to lose access to anything already stored in `temporaryDataStorage`.

This is an edge case that I don't think was really possible, but I'd rather be explicit about it just in case I'm missing some scenario in which this might happen